### ISSUE TITLE
About I3D test accuracy and confusion matrix

### ISSUE THREAD

**Issue by ENDLICHERE** [dxli94/WLASL#22]
Hello, 
 I find that in test_i3d.py, the top-k accuracy are calculated as TP/(TP+FP), while the general recognition accuracy might be  (TP+TN)/(TP+FP+TN+FN). 
In train_i3d.py, the accuracy are calculated as `acc = float(np.trace(confusion_matrix)) / np.sum(confusion_matrix)`, so I bulid the confusion_matrix in test_i3d.py too, and get acc = 0.0. (Using WLASL100, and the given weight FINAL_nslt_100_iters=896_top1=65.89_top5=84.11_top10=89.92.pt)
I wonder why this happened, or is there something wrong with my implement?

Start from line 104 in test_i3d.py:
`    for data in dataloaders["test"]:
        inputs, labels, video_id = data  # inputs: b, c, t, h, w

        per_frame_logits = i3d(inputs)

        predictions = torch.max(per_frame_logits, dim=2)[0]
        out_labels = np.argsort(predictions.cpu().detach().numpy()[0])
        out_probs = np.sort(predictions.cpu().detach().numpy()[0])

        confusion_matrix[labels.item(), out_labels[0]] += 1

        if labels[0].item() in out_labels[-5:]:
            correct_5 += 1
            top5_tp[labels[0].item()] += 1
        else:
            top5_fp[labels[0].item()] += 1
        if labels[0].item() in out_labels[-10:]:
            correct_10 += 1
            top10_tp[labels[0].item()] += 1
        else:
            top10_fp[labels[0].item()] += 1
        if torch.argmax(predictions[0]).item() == labels[0].item():
            correct += 1
            top1_tp[labels[0].item()] += 1
        else:
            top1_fp[labels[0].item()] += 1
        print(video_id, float(correct) / len(dataloaders["test"]), float(correct_5) / len(dataloaders["test"]),
              float(correct_10) / len(dataloaders["test"]))

        # per-class accuracy
    top1_per_class = np.mean(top1_tp / (top1_tp + top1_fp))
    top5_per_class = np.mean(top5_tp / (top5_tp + top5_fp))
    top10_per_class = np.mean(top10_tp / (top10_tp + top10_fp))
    print('top-k average per class acc: {}, {}, {}'.format(top1_per_class, top5_per_class, top10_per_class))

    np.save('confusion_matrix.npy', confusion_matrix)
    acc = float(np.trace(confusion_matrix)) / np.sum(confusion_matrix)
    print('acc= ', acc)`
